### PR TITLE
add user permissions

### PR DESCRIPTION
**Pull request by vasylkushnir**

### CODE REVIEW

File: src/models/users.model.ts [hunk 1]
```diff
@@ -4,6 +4,10 @@ import { Sequelize, DataTypes, Model } from 'sequelize';
 import { Application } from '../declarations';
 import { HookReturn } from 'sequelize/types/lib/hooks';
 
+export enum UserPermissions {
```
Comment (m-ilchuk) [inline src/models/users.model.ts]: That's not permissions that's `Roles`.

File: src/models/users.model.ts [hunk 2]
```diff
@@ -4,6 +4,10 @@ import { Sequelize, DataTypes, Model } from 'sequelize';
...
+  ADMIN ='ADMIN',
```
Comment (m-ilchuk) [inline src/models/users.model.ts:9]: Add SUPER_ADMIN too.

File: src/services/notifications/notifications.hooks.ts [hunk 3]
```diff
@@ -54,6 +55,9 @@ export default {
     ],
     get: [isValidId(notificationId)],
     create: [
+      checkPermissions({
+        roles: [ 'ADMIN' ],
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.hooks.ts]: Use created enum. ADMIN or SUPER_ADMIN

File: src/services/users/users.hooks.ts [hunk 4]
```diff
@@ -17,6 +18,9 @@ export default {
     all: [],
     find: [
       authenticate('jwt'),
+      checkPermissions({
```
Comment (m-ilchuk) [inline src/services/users/users.hooks.ts:34]: same ^

**Review by m-ilchuk — CHANGES_REQUESTED** [review]

File: src/models/users.model.ts (hunk 1 above)
Comment (vasylkushnir) [reply src/models/users.model.ts]: changed bd58a083101312d0e3c7512ef88955c17798aeec

File: src/models/users.model.ts (hunk 2 above)
Comment (vasylkushnir) [reply src/models/users.model.ts:9]: added bd58a083101312d0e3c7512ef88955c17798aeec

File: src/services/notifications/notifications.hooks.ts (hunk 3 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.hooks.ts]: enum added bd58a083101312d0e3c7512ef88955c17798aeec

File: src/services/users/users.hooks.ts (hunk 4 above)
Comment (vasylkushnir) [reply src/services/users/users.hooks.ts:34]: chenged bd58a083101312d0e3c7512ef88955c17798aeec

File: src/hooks/isCurrent.ts [hunk 5]
```diff
@@ -8,7 +9,8 @@ import { Hook, HookContext } from '@feathersjs/feathers';
  */
 export default (): Hook => {
   return (context: HookContext): HookContext => {
-    if (context.params.user?.id !== context.id) {
+    const user = context.params.user?.permissions;
+    if ((!user.includes(UserRoles.ADMIN)) && (context.params.user?.id !== context.id)) {
```
Comment (m-ilchuk) [inline src/hooks/isCurrent.ts]: Please, remove not needed extra brackets

File: src/services/notifications/notifications.hooks.ts [hunk 6]
```diff
@@ -54,6 +56,9 @@ export default {
     ],
     get: [isValidId(notificationId)],
     create: [
+      checkPermissions({
+        roles: [ UserRoles.ADMIN ],
```
Comment (m-ilchuk) [inline src/services/notifications/notifications.hooks.ts]: or SUPER_ADMIN

File: src/services/users/users.hooks.ts [hunk 7]
```diff
@@ -32,13 +49,15 @@ export default {
     update: [
       authenticate('jwt'),
       isValidId(userId),
+      iff(isProvider('external'),setUserRole),
```
Comment (m-ilchuk) [inline src/services/users/users.hooks.ts]: space

File: src/services/users/users.hooks.ts [hunk 8]
```diff
@@ -32,13 +49,15 @@ export default {
...
       validate.form(replaceUserSchema),
       iff(isProvider('external'), isCurrent(), isUniqueEmail()),
       hashPassword('password'),
     ],
     patch: [
       authenticate('jwt'),
       isValidId(userId),
+      iff(isProvider('external'),setUserRole),
```
Comment (m-ilchuk) [inline src/services/users/users.hooks.ts]: same ^

**Review by m-ilchuk — CHANGES_REQUESTED** [review]

File: src/services/users/users.hooks.ts [hunk 9]
```diff
@@ -7,16 +7,33 @@ import isCurrent from '../../hooks/isCurrent';
...
+import checkPermissions from 'feathers-permissions';
+import { UserRoles } from '../../models/users.model';
+import { HookContext } from '@feathersjs/feathers';
 // Don't remove this comment. It's needed to format import lines nicely.
 
 const { authenticate } = feathersAuthentication.hooks;
 const { hashPassword, protect } = local.hooks;
 
+function setUserRole(context: HookContext): HookContext {
+  const user = context.params.user?.permissions;
```
Comment (m-ilchuk) [inline src/services/users/users.hooks.ts]: const permissions = context.params.user?.permissions; 

or 

const { permissions } = context.params.user;

**Review by m-ilchuk — CHANGES_REQUESTED** [review]

File: src/hooks/isCurrent.ts (hunk 5 above)
Comment (vasylkushnir) [reply src/hooks/isCurrent.ts]: removed cc2de6c4baad0c9ed12205e453a9abca8ed0fdf7

File: src/services/notifications/notifications.hooks.ts (hunk 6 above)
Comment (vasylkushnir) [reply src/services/notifications/notifications.hooks.ts]: added cc2de6c4baad0c9ed12205e453a9abca8ed0fdf7

File: src/services/users/users.hooks.ts (hunk 7 above)
Comment (vasylkushnir) [reply src/services/users/users.hooks.ts]: added cc2de6c4baad0c9ed12205e453a9abca8ed0fdf7

File: src/services/users/users.hooks.ts (hunk 8 above)
Comment (vasylkushnir) [reply src/services/users/users.hooks.ts]: added cc2de6c4baad0c9ed12205e453a9abca8ed0fdf7

File: src/services/users/users.hooks.ts (hunk 9 above)
Comment (vasylkushnir) [reply src/services/users/users.hooks.ts]: changed cc2de6c4baad0c9ed12205e453a9abca8ed0fdf7

**Pull request merged**